### PR TITLE
M2 practice lab 3

### PR DESCRIPTION
**Pull request by javasquezgo**
In this lab we were doing SQL queries with Rails, it was something in which I felt more comfortable. My area stands out with SQL, so for the moment I am moving forward, this branch contains everything. I started from scratch due to the fact that I had not understood making the models within rails and that was what took me the most time since I came loaded with lab 2 that was this week but right now everything is up to date.

### CODE REVIEW

File: app/models/retweet.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  validates :reply_text, length: {maximum:255}
+
+  #Retweets counts: Create a new scope that retrieves the number of retweets
+  scope :count_of_retweet, ->(user_id) {
+    joins("JOIN retweets r ON users.id = r.user_id")
+      .joins("LEFT JOIN tweets t ON t.id = r.tweet_id")
+      .select("u.user_name, t.content, r.reply_text")
+      .where("users.id = ?", user_id)
+  }
```
Comment (ragar90) [inline app/models/retweet.rb:13]: Why are you joining it with itself? don't you think is unnecessary? Also why using LEFT JOIN don't you thinkit would be better to use inner join? with left join you also are bringing the tweets without retweets, also you forgot .count

File: app/models/retweet.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  def self.retweeted_tweets(user_id)
+    select('retweets.reply_text, tweets.content')
+      .joins('JOIN retweets ON users.id = retweets.user_id')
+      .joins('LEFT JOIN tweets ON tweets.id = retweets.tweet_id')
+      .where('users.id = ?', user_id)
+  end
```
Comment (ragar90) [inline app/models/retweet.rb:20]: Same thing here, is not necessary to join it with itself

**Review by ragar90 — APPROVED** [review]
Seems you need to review your SQL knowledge o probably rethink your model to improve your Queries

**Comment by RodSoriano** [comment]
No pull-request should be opened on the main project. All forks needs to be within the fork of each trainee.

**Pull request closed without merging**